nautobot/ui/src/views/generic/GenericView.js

```diff
@@ -0,0 +1,247 @@
+import {
+    AutomationIcon,
+    Box,
+    Breadcrumb,
+    Breadcrumbs,
+    DcimIcon,
+    Divider,
+    Flex,
+    Input,
+    InputGroup,
+    InputLeftElement,
+    IpamIcon,
+    Menu,
+    MenuButton,
+    MenuItem,
+    MenuList,
+    NautobotGrid,
+    Navbar,
+    NavbarMenuButton,
+    NavbarNotificationButton,
+    NavbarSection,
+    NavbarSections,
+    negateCssVar,
+    PlatformIcon,
+    Popover,
+    PopoverArrow,
+    PopoverBody,
+    PopoverContent,
+    PopoverTrigger,
+    SearchIcon,
+    SecurityIcon,
+    Text,
+} from "@nautobot/nautobot-ui";
+import { useMemo } from "react";
+import {
+    Link as ReactRouterLink,
+    NavLink as ReactRouterNavLink,
+    useLocation,
+} from "react-router-dom";
+
+import { useGetSessionQuery, useGetUIMenuQuery } from "@utils/api";
+
+export default function GenericView({
+    children,
+    columns = "4",
+    objectData,
+    rows,
+}) {
+    const { pathname } = useLocation();
+
+    const menu = useGetUIMenuQuery();
+    const session = useGetSessionQuery();
+
+    const breadcrumbs = useMemo(
+        () =>
+            (menu?.data ?? []).reduce((breadcrumbs, root) => {
```

Comment:
the data structure created here is very component-specific, so I wouldn't move it anywhere outside. if so required, perhaps it would be a good idea to split it into functions returning one menu item and/or breadcrumb at a time, e.g. `getUIMenuItem(pathname, level)`, `getBreadcrumb(uiMenuItem)`, and composing according to specific requirements. also, enabling breadcrumbs overrides through `GenericView` props of course, but that's the simplest thing to do. would that work for you?